homeassistant/components/ezviz/camera.py

```diff
@@ -0,0 +1,276 @@
+    }
+)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the Ezviz IP Cameras."""
+    conf_cameras = {}
+    if CONF_CAMERAS in config:
+        conf_cameras = config[CONF_CAMERAS]
+        _LOGGER.debug("Expecting %s cameras from config", len(conf_cameras))
+
+    account = config[CONF_USERNAME]
+    password = config[CONF_PASSWORD]
+
+    try:
+        ezviz_client = EzvizClient(account, password)
+        ezviz_client.login()
+
+        # Get cameras
+        connections = ezviz_client.get_CONNECTION()
+        devices = ezviz_client.get_DEVICE()
+        # now, let's build the HASS devices
+        cameras = {}
+        for device in devices:
+
+            device_serial = device["deviceSerial"]
+
+            # There seem to be a bug related to localRtspPort in Ezviz API...
+            local_rtsp_port = 554
+            if connections[device_serial]["localRtspPort"] != 0:
+                local_rtsp_port = connections[device_serial]["localRtspPort"]
+
+            cameras[device_serial] = {
+                "serial": device_serial,
+                "name": device["name"],
+                "deviceType": device["deviceType"],
+                "version": device["version"],
+                "status": device["status"],
+                "createTime": device["userDeviceCreateTime"],
+                "category": device["deviceCategory"],
+                "subCategory": device["deviceSubCategory"],
+                "customType": device["customType"],
+                "localIp": connections[device_serial]["localIp"],
+                "netIp": connections[device_serial]["netIp"],
+                "localRtspPort": local_rtsp_port,
+                "netType": connections[device_serial]["netType"],
+                "wanIp": connections[device_serial]["wanIp"],
+            }
+
+        # Add the cameras as devices in HASS
+        for camera_serial in cameras:
+            camera = cameras[camera_serial]
+            _LOGGER.debug("CAMERA: %s", camera)
+
+            camera_username = CONF_DEFAULT_CAMERA_USERNAME
+            camera_password = ""
+            camera_rtsp_stream = ""
+
+            if camera_serial in conf_cameras:
+                camera_username = conf_cameras[camera_serial]["username"]
+                camera_password = conf_cameras[camera_serial]["password"]
+                camera_rtsp_stream = "rtsp://{}:{}@{}:{}".format(
+                    camera_username,
+                    camera_password,
+                    camera["localIp"],
+                    camera["localRtspPort"],
+                )
+                _LOGGER.debug(
+                    "Camera %s source stream: %s", camera["serial"], camera_rtsp_stream
+                )
+
+            else:
+                _LOGGER.error(
+                    "I found a camera (%s) but it is not configured. Please configure it if you wish to see the appropriate stream. Conf cameras: %s",
```

Comment:
Please keep the language conservative in the log messages. Don't use first person grammar.